llvm/docs/LangRef.rst

```diff
@@ -4426,6 +4425,21 @@ represented by ``0xH`` followed by 4 hexadecimal digits. The bfloat 16-bit
 format is represented by ``0xR`` followed by 4 hexadecimal digits. All
 hexadecimal formats are big-endian (sign bit at the left).
 
+Some of the special floating point values can be represented by the following
+identifiers:
+
+    +-----------+---------------------------------------------------+
+    | Name      | Description                                       |
+    +===========+===================================================+
+    | ``qnan``  | Positive quiet NaN w/ payload equals to zero      |
+    +-----------+---------------------------------------------------+
+    | ``snan``  | Positive signaling NaN w/ payload equals to zero  |
```

Comment:
```suggestion
    | ``snan``  | Positive signaling NaN w/ payload equal to zero   |
```

Reply:
There is no sNaN with an all-0 payload, that would be an infinity instead--the payload needs to have some bit set to not be an infinity. qNaNs always have one bit set, but there is no requisite bit set for the sNaN.

Reply:
Yeah it's a bit tricky to describe what we want here...maybe we could just say "value equal to APFloat::getSNaN()"? Though I'm not sure if we should rely on an API, which is subject to change anytime, to describe a specification. What do you think?

Reply:
Maybe it should just be explicit about the payload, e.g. `nan(payload)` instead of `qnan`/`snan`.

Reply:
I haven't looked at how hard the logic would be to implement, but making `nan` by itself return the preferred qNaN and `nan(payload)` be used for other qNaN/sNaN payloads seems the best way forward to me.